source/1_engine/rex_engine/include/rex_engine/diagnostics/logging/log_category.h

```diff
@@ -2,38 +2,48 @@
 
 #include "rex_engine/diagnostics/logging/log_verbosity.h"
 #include "rex_std/string.h"
+#include "rex_std_extra/utility/yes_no.h"
 
 namespace rex
 {
-  using LogCategoryName = rsl::string_view;
+    DEFINE_YES_NO_ENUM(IsAsync);
 
-  /** Base class for all log categories. **/
-  struct LogCategory
-  {
-    /**
-     * Constructor, registers with the log suppression system and sets up the default values.
-     * @param CategoryName, name of the category
-     * @param InDefaultVerbosity, default verbosity for the category, may ignored and overridden by many other mechanisms
-     **/
-    LogCategory(const LogCategoryName& inCategoryName, LogVerbosity inDefaultVerbosity);
+    using LogCategoryName = rsl::string_view;
 
-    /** Gets the category name **/
-    const LogCategoryName& get_category_name() const;
+    /** Base class for all log categories. **/
+    struct LogCategory
+    {
+        /**
+         * Constructor, registers with the log suppression system and sets up the default values.
+         * @param CategoryName, name of the category
+         * @param InDefaultVerbosity, default verbosity for the category, may ignored and overridden by many other mechanisms
+         * @param inAsync, should this Category log async or not.
+         **/
+        LogCategory(const LogCategoryName& inCategoryName, LogVerbosity inDefaultVerbosity, IsAsync inAsync = IsAsync::no);
 
-    /** Gets the working verbosity **/
-    const LogVerbosity& get_verbosity() const;
+        /** Gets the category name **/
+        const LogCategoryName& get_category_name() const;
 
-    /** Should not generally be used directly. Tests the runtime verbosity and maybe triggers a debug break, etc. **/
-    bool is_suppressed(LogVerbosity verbosityLevel) const;
+        /** Gets the working verbosity **/
+        const LogVerbosity& get_verbosity() const;
```

Comment:
doesn't have to return by const ref, it's an enum

Reply:
It does not but I do it to be consistent with most other custom types, I'll adjust it